File: frontend/types/memory.ts
export type MemoryKind = "fact" | "note" | "preference" | "context";

export interface MemoryRecord {
  memory_id: string;
  agent_id: string;
  content: string;
  kind: MemoryKind;
  created_at: string;
  updated_at: string;
  metadata: Record<string, unknown>;
  tags: string[];
}

export interface MemoryListResponse {
  memories: MemoryRecord[];
}

export interface CreateMemoryInput {
  content: string;
  kind: MemoryKind;
  metadata: Record<string, unknown>;
  tags: string[];
}

export interface UpdateMemoryInput {
  content?: string;
  kind?: MemoryKind;
  metadata?: Record<string, unknown>;
  tags?: string[];
}

File: backend/app/core/api/schemas/memory.py
"""Pydantic contracts for the Memory Runtime API."""

from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field
from backend.app.core.memory.domain.memory_kind import MemoryKind
from backend.app.core.memory.domain.memory_record import MemoryRecord

class MemoryResponse(BaseModel):
    memory_id: UUID; agent_id: UUID; content: str; kind: MemoryKind; created_at: datetime; updated_at: datetime; metadata: dict[str, object]; tags: list[str]
    @classmethod
    def from_memory(cls, memory: MemoryRecord) -> "MemoryResponse": return cls(memory_id=memory.memory_id, agent_id=memory.agent_id, content=memory.content, kind=memory.kind, created_at=memory.created_at, updated_at=memory.updated_at, metadata=dict(memory.metadata), tags=list(memory.tags))
class MemoryListResponse(BaseModel): memories: list[MemoryResponse]
class MemoryCreateRequest(BaseModel): content: str = Field(min_length=1, max_length=10_000); kind: MemoryKind = MemoryKind.NOTE; metadata: dict[str, object] = Field(default_factory=dict); tags: list[str] = Field(default_factory=list)
class MemoryUpdateRequest(BaseModel): content: str | None = Field(default=None, min_length=1, max_length=10_000); kind: MemoryKind | None = None; metadata: dict[str, object] | None = None; tags: list[str] | None = None
